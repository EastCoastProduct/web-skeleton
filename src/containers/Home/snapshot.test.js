import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { HomeComponent } from './';

describe('Home component snapshot', () => {
  it('render without confirm message', () => {
    const tree = renderer.create(
      <HomeComponent
        auth={fromJS({ emailResendError: null, emailResendSuccess: false })}
        dispatch={() => {}}
        user={fromJS({ confirmed: true })}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render with confirm message', () => {
    const tree = renderer.create(
      <HomeComponent
        auth={fromJS({ emailResendError: null, emailResendSuccess: false })}
        dispatch={() => {}}
        user={fromJS({ confirmed: false })}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render with confirm message and email resent', () => {
    const tree = renderer.create(
      <HomeComponent
        auth={fromJS({ emailResendError: null, emailResendSuccess: true })}
        dispatch={() => {}}
        user={fromJS({ confirmed: false })}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render with confirm message and email resend error', () => {
    const error = 'Something went wrong';
    const tree = renderer.create(
      <HomeComponent
        auth={fromJS({ emailResendError: error, emailResendSuccess: false })}
        dispatch={() => {}}
        user={fromJS({ confirmed: false })}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
