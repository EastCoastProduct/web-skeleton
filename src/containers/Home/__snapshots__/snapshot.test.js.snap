exports[`Home component snapshot render with confirm message 1`] = `
<main>
  <div>
    <p>
      You have to confirm your email. Please open email to follow link or 
      <a
        href={true}
        onClick={[Function]}>
        resend
      </a>
       email.
    </p>
  </div>
  <p>
    Home Page
  </p>
</main>
`;

exports[`Home component snapshot render with confirm message and email resend error 1`] = `
<main>
  <div>
    <p>
      You have to confirm your email. Please open email to follow link or 
      <a
        href={true}
        onClick={[Function]}>
        resend
      </a>
       email.
    </p>
    <p>
      <i
        className="fa fa-exclamation-triangle" />
      Something went wrong
    </p>
  </div>
  <p>
    Home Page
  </p>
</main>
`;

exports[`Home component snapshot render with confirm message and email resent 1`] = `
<main>
  <div>
    <p>
      You have to confirm your email. Please open email to follow link or 
      <a
        href={true}
        onClick={[Function]}>
        resend
      </a>
       email.
    </p>
    <p>
      Email sent!
    </p>
  </div>
  <p>
    Home Page
  </p>
</main>
`;

exports[`Home component snapshot render without confirm message 1`] = `
<main>
  <p>
    Home Page
  </p>
</main>
`;
